bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/ConfigResolver.java

```diff
@@ -46,100 +48,137 @@
+                        return getConfigValue(module, key, configProvider -> configProvider
                                 .getAsRecordAndMark(module, key));
                     case TypeTags.TABLE_TAG:
-                        return getConfigValue(configProviders, configProvider -> configProvider
+                        return getConfigValue(module, key, configProvider -> configProvider
                                 .getAsTableAndMark(module, key));
+                    case TypeTags.XML_TAG:
+                    case TypeTags.XML_ELEMENT_TAG:
+                    case TypeTags.XML_COMMENT_TAG:
+                    case TypeTags.XML_PI_TAG:
+                    case TypeTags.XML_TEXT_TAG:
+                        return getConfigValue(module, key, configProvider -> configProvider
+                                .getAsXmlAndMark(module, key));
                     default:
                         throw new ErrorValue(StringUtils.fromString(
                                 String.format(CONFIGURATION_NOT_SUPPORTED, key.module.getName() + ":" + key.variable,
                                               effectiveType.toString())));
                 }
             default:
-                throw new ErrorValue(StringUtils.fromString(
-                        String.format(CONFIGURATION_NOT_SUPPORTED, key.module.getName() + ":" + key.variable,
-                                      type.toString())));
+                diagnosticLog.error(String.format(CONFIGURATION_NOT_SUPPORTED,
+                                                  key.module.getName() + ":" + key.variable, type.toString()));
         }
+        return Optional.empty();
     }
 
-    private Optional<?> getConfigValue(List<ConfigProvider> configProviders,
-                                          Function<ConfigProvider, Optional<?>> getConfigFunc) {
+    private Optional<?> getConfigValue(Module module, VariableKey key,
+                                       Function<ConfigProvider, Optional<?>> getConfigFunc) {
         Optional<?> configValue = Optional.empty();
-        for (ConfigProvider configProvider : configProviders) {
-            Optional<?> value = getConfigFunc.apply(configProvider);
-            if (value.isPresent()) {
-                 configValue = value;
+        List<ConfigException> exceptionList = new ArrayList<>(runtimeConfigProviders.size());
+        for (ConfigProvider configProvider : runtimeConfigProviders) {
+            try {
+                Optional<?> value = getConfigFunc.apply(configProvider);
+                if (configValue.isEmpty()) {
+                    configValue = value;
+                }
+            } catch (ConfigException e) {
+                exceptionList.add(e);
+            }
+        }
+
+        // Handle errors while getting config values.
+        if (configValue.isPresent()) {
+            exceptionList.forEach(e -> diagnosticLog.warn(e.getMessage()));
```

Comment:
This is the logic we check whether we print war or error. If at least one providers has a value we print them as warnings.